### PR TITLE
min max wrapper

### LINKED ISSUE #49: MinMaxMetric for wrapping other metrics
### Motivation

- `MinMaxMetric` is a metric that simply wraps another metric (e.g.`val_acc`) and creates a new metric that tracks the `min`, `max`  or both values of `val_acc`. 

### Pitch

- I personally use it to quickly see the `max_val_acc` of a complete experiment in TensorBoard (instead of going through the graph manually to find the max value) but I can see other usecases as well. 
- It was discussed in the PL Slack [here](https://pytorch-lightning.slack.com/archives/CU5PF2E2J/p1614602988020800) and clearly resonated with more users

### Additional context
- Happy to submit a PR for this feature, as I do have already a (incomplete) `MaxMetric` code implemented [here](https://gist.github.com/janhenriklambrechts/67a0faf5dc05d4e8d4d16973d1c03929)

### PR DESCRIPTION
**Pull request by janhenriklambrechts**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?

Fixes #49

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@SkafteNicki seems to be a duplicate to #506

**Comment by Borda** [comment]
@janhenriklambrechts how is it going here, ready for review?

**Comment by janhenriklambrechts** [comment]
@Borda not yet finished, but will work on it this afternoon -- so expect a clean PR by tomorrow :)

**Comment by Borda** [comment]
@SkafteNicki mind have look ^^ :rabbit:

File: tests/wrappers/test_minmax.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+def compare_fn_ddp(preds, target, base_fn):
+    min, max = 1e6, -1e6  # pick some very large numbers for comparing
+    for i, j in zip(range(0, NUM_BATCHES, 2), range(1, NUM_BATCHES, 2)):
+        p = torch.cat([preds[i * BATCH_SIZE : (i + 1) * BATCH_SIZE], preds[j * BATCH_SIZE : (j + 1) * BATCH_SIZE]])
+        t = torch.cat([target[i * BATCH_SIZE : (i + 1) * BATCH_SIZE], target[j * BATCH_SIZE : (j + 1) * BATCH_SIZE]])
+        base_fn.update(p, t)
+        val = base_fn.compute().cpu().numpy()
+        min = min if min < val else val
+        max = max if max > val else val
+        print(min, max)
```
Comment (SkafteNicki) [inline tests/wrappers/test_minmax.py:48]:
```suggestion
```
